src/markdown.c

```diff
@@ -769,8 +769,33 @@ char_entity(struct buf *ob, struct sd_markdown *rndr, uint8_t *data, size_t max_
 		if (!is_valid_numeric_entity(entity_val))
 			return 0;
 	} else {
-		if (!is_allowed_named_entity((const char *)data, end))
+		/* Lookup the entity in the named entity table. */
+		const struct html_entity *entity = is_allowed_named_entity((const char *)data, end);
+		if (!entity)
 			return 0;
+		if (entity->codepoints[0]) {
+			/* Convert the entity to numeric entities. */
+			size_t entitystr_size = MAX_NUM_ENTITY_LEN+10;
```

Comment:
What does the `+10` mean? Seems like we need room for (the longest-possible decimal entity * MAX_ENTITY_CODEPOINTS) + 1